framework/src/controller/controller.js

```diff
@@ -123,15 +137,21 @@ class Controller {
 
 		this.channel = new EventEmitterChannel(
 			'lisk',
-			['ready'],
+			['ready', 'applicationStateUpdated'],
 			{
 				getComponentConfig: action => this.componentConfig[action.params],
+				getApplicationState: () => this.applicationState.getState(),
+				updateApplicationState: action =>
+					this.applicationState.update(action.params),
 			},
 			this.bus,
 			{ skipInternalEvents: true }
 		);
+
 		await this.channel.registerToBus();
 
+		await this.applicationState.setChannel(this.channel);
```

Comment:
should it have access to channel? maybe to `bus` instead?

Reply:
Why bus? I think channel is the correct.